l10n_br_purchase/views/purchase_view.xml

```diff
@@ -0,0 +1,95 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+      <!--busca por cnpj_cpf e razao social nos pedidos de compra -->
+      <record id="l10n_br_purchase_order_filter" model="ir.ui.view">
+          <field name="name">l10n_br_purchase.partner.filter</field>
+          <field name="model">purchase.order</field>
+          <field name="inherit_id" ref="purchase.view_purchase_order_filter"/>
+          <field name="arch" type="xml">
+              <field name="name" position="after">
+                  <field name="legal_name" string="Razão Social"/>
+                  <field name="cnpj_cpf" string="CNPJ/CPF"/>
+                  <field name="ie" string="Inscr. Estadual"/>
+              </field>
+          </field>
+      </record>
+
+      <record id="l10n_br_purchase_order_form" model="ir.ui.view">
+          <field name="name">l10n_br_purchase.order.form</field>
+          <field name="model">purchase.order</field>
+          <field name="inherit_id" ref="purchase.purchase_order_form"/>
+          <field name="priority">36</field>
+          <field name="arch" type="xml">
+              <xpath expr="//sheet/div[@name='button_box']/button" position="after">
+                <button type="object" name="action_view_document" class="oe_stat_button" icon="fa-pencil-square-o" attrs="{'invisible':['|', ('invoice_count', '=', 0), ('state', 'in', ('draft','sent','to approve'))]}">
+                    <field name="fiscal_document_count" widget="statinfo" string="Documents"/>
+                </button>
+              </xpath>
+              <xpath expr="//field[@name='company_id']" position="after">
+                  <field name="fiscal_operation_id" required="1"/>
+              </xpath>
+              <xpath expr="//field[@name='order_line']" position="attributes">
+                  <attribute name="context">{'default_fiscal_operation_id': fiscal_operation_id, 'default_company_id': company_id}</attribute>
+              </xpath>
+              <!-- Edit order_line Tree view -->
+              <xpath expr="//field[@name='order_line']/tree" position="attributes">
+                  <attribute name="editable"/>
+              </xpath>
+              <xpath expr="//field[@name='order_line']/tree/field[@name='taxes_id']" position="attributes">
+                  <attribute name="invisible">1</attribute>
+              </xpath>
+              <xpath expr="//field[@name='order_line']/tree/field[@name='taxes_id']" position="before">
+                  <field name="fiscal_tax_ids" widget="many2many_tags"/>
+              </xpath>
+              <xpath expr="//field[@name='order_line']/tree/field[@name='company_id']" position="replace"/>
```

Comment:
essa extensão xpath vazia nao seria inútil?

Reply:
Eh para remover o company_id da tree view da linha das ordens de compras, mas eu acho que seria melhor deixa-lo invisible="1"

Reply:
@rvalyi fiz o push para melhorar esse xpath...

Reply:
porque remove-la? Isso não seria o tipo de deciçao para se tomar ao nível da akretion/usability sera se tem que ser o papel o l10n_br_sale de tomar esse tipo de opinião? Tem uma justificativa profunda ou é so que vc acha a usabilidade melhor assim? Porque se o motivo for esse, ai sim melhor no akretion/usability ou modulo do tipo...

Reply:
o campo original esta assim
`<field name="company_id" groups="base.group_multi_company" options="{'no_create': True}"/>`
ou seja, vc so ve se tiver no grupo group_multi_company, na v12, v13 e v14. Ou seja, é mais para ser feature do que bug.

Eu imagino que isso seria para se uma empresa vai comprar produtos para varias outras (matriz e filial por examplo?). Caso isso pegar em algum cliente nosso que tem group_multi_company eu diria que deveriamos esconder o campo no akretion/usability ou no modulo do cliente, mas nao aqui na localizaçao. O que vc acha?

Reply:
Eu deixo atualmente invisível porque na localização o pedido de compras sempre será de uma empresa e hoje colocamos outros campos então deixei esse campo invisível para ter mais espaço na tela

Reply:
blz, faz sentido se na localizaçao tem que ser da mesma empresa mesmo ok.

Reply:
Aproveitando a conversa.. vcs tem alguma situação de empresa com 2 cnpjs do simples e que somente na hora do faturamento decide qual empresa será feito a emissão da NFe ou NFSe ?

Reply:
@marcelsavegnago a gente nao teve clientes assim. Mas pensei que poderia existir. Se vc ver uma forma simples du suportar o caso é bem vindo.

Reply:
@rvalyi cliente cliente não temos tbm mas em conversa com alguns prospecto foi comentado este tipo de necessidade. Não acho que seja algo simples principalmente se considerar tudo o que foi gerado para atender esta venda (ordem de produção, compras, pickings e afins)

Reply:
ai nesse caso, vale a pena fazer com modulos customizados. Felizmente como a gente sempre chama o super e tem so metodo pequenos, a principio vc consegue contornar tudo que nao gostar em casos mais cabeludos.